### ISSUE TITLE
About coding style, do we need a clang-format configuration

### ISSUE THREAD

**Issue by RuiwenTang** [thorvg/thorvg#1519]
As mentioned in the contribution guidelines, the coding style check must passed. 
So I think a clang-format file with specified style config is helpful.

By the way, I tried the code_format.sh script, but the result seems not consistent with the existing code

**Comment by capnm** [comment]
++ EOL issues. I see that by converting and commiting the ThorVG code to the Godot github repository.
See https://github.com/godotengine/godot/blob/master/.gitattributes

**Comment by hermet** [comment]
I agree though I don't like to change the tvg coding style at this point since it will hurts the usability of the code history tracking like git blame info...  

code_format.sh has been experimentally added in the past, we need to improve it to work properly or renewal the official tvg codiing style at the 1.0 release with a more popular one? I just ignore this task since it's annoying to me so far. A volunteer is welcomed. :-)

**Comment by capnm** [comment]
> I agree though I don't like to change the tvg coding style at this point since it will hurts the usability of the code history tracking like git blame info...

Yes, reformatting code also hides usable line-blame annotations in code editors. Maybe just keep some visual consistency that is human readable;)

**Comment by lmdsp** [comment]
I noticed this style pattern is used mostly everywhere for conditions:

Result draw()
{
    if (drawing || paints.empty() || !renderer || !renderer->preRender()) return Result::InsufficientCondition;

    bool rendered = false;
    for (auto paint : paints) {
        if (paint->pImpl->render(renderer)) rendered = true;
    }

    if (!rendered || !renderer->postRender()) return Result::InsufficientCondition;

    drawing = true;

    return Result::Success;
}

This makes it impossible to set breakpoints for debugging.
May I suggest doing something like this instead

Result draw()
{
    if (drawing || paints.empty() || !renderer || !renderer->preRender())
    {
       return Result::InsufficientCondition; // breakpoint can be set here
    }

    bool rendered = false;
    for (auto paint : paints) {
        if (paint->pImpl->render(renderer))
        {
          rendered = true;
        }
    }

    if (!rendered || !renderer->postRender())
    {
      return Result::InsufficientCondition; // breakpoint can be set here
    }

    drawing = true;

    return Result::Success;
}
